src/wp-includes/functions.php

```diff
@@ -8333,3 +8333,35 @@ function is_php_version_compatible( $required ) {
 function wp_fuzzy_number_match( $expected, $actual, $precision = 1 ) {
 	return abs( (float) $expected - (float) $actual ) <= $precision;
 }
+
+/**
+ * Replaces periods with encoded entities in URLs.
+ *
+ * If the URL ends with a period, replace it with %2E to avoid
+ * misinterpretation from some email clients.
+ *
+ * @since 5.9.0
+ *
+ * @param string $url The URL to encode.
+ * @return string The new encoded URL.
+ */
+function wp_url_encode_ending_period( $url ) {
```

Comment:
If this function remains, it should be a core-only private function. And it should named something like `_encode_query_arg_ending_period()` to better describe what it's doing while not being named for `rawurlencode()`.